packages/playwright-core/src/client/android.ts

```diff
@@ -244,6 +246,7 @@ export class AndroidDevice extends ChannelOwner<channels.AndroidDeviceChannel> i
         this._connection.close();
       else
         await this._channel.close();
+      await this._closedPromise;
```

Comment:
Could you provide an example where this is necessary? The snippet in the description patches Playwright methods, so we can't use it as a justification.